bb.edn

```diff
@@ -0,0 +1,33 @@
+{:paths ["src"]
+ :tasks
+ {:requires [[babashka.deps :as deps]
+             [babashka.fs :as fs]
+             [babashka.process :as p]
+             [clojure.string :as str]]
+
+  lint {:doc "Run linter."
+        :task (do (when-not (fs/exists? ".clj-kondo")
+                    ;; Prep linter. The `:test` alias has main as dep.
+                    (let [cp (with-out-str (deps/clojure ["-A:test" "-Spath"]))]
+                      (fs/create-dir ".clj-kondo")
+                      (p/shell "clj-kondo" "--lint" cp "--dependencies" "--copy-configs")))
+                  (doseq [dir ["src" "test"]]
+                    (println :linting dir "...")
+                    (p/shell "clj-kondo" "--lint" dir)
+                    (println)))}
```

Comment:
We could use clojure-lsp itself to lint :) We would have clean-ns and format additional tasks
Not blocker for this PR though

Reply:
Indeed! It didn't cross my mind that you can use clojure-lsp as a command line linter, this is a really great!

I've tried it and it doesn't work on MS-Windows yet, there are a lot of false positives reported; these are fixed in the clojure-lsp patches I've provided recently pending a release.

Btw, how would you about handling definitions that are not reference anywhere, even in tests (this is on linux)?

``` shell
~/src/deps-bin $ bb lint
:linter diagnostics ...
[ 20%] Copying kondo configs       Configs copied:
- .clj-kondo/babashka/fs
- .clj-kondo/rewrite-clj/rewrite-clj
[ 99%] Analyzing project files     No configs copied.
[100%] Project analyzed            
Finding diagnostics...
src/deps_bin/deps_bin.clj:4:6: info: [clojure-lsp/unused-public-var] Unused public var 'deps-bin.deps-bin/bin'
```

It's a simple change to switch to clojure-lsp, and I've noticed you have developed a GH action https://github.com/clojure-lsp/setup-clojure-lsp/tree/main/src too :)

bb.edn change
``` clojure
lint {:doc "Run clojure-lsp linter."
        :task (doseq [linter '[diagnostics]]
                (println :linter linter "...")
                (p/shell "clojure-lsp" linter "--dry" "--project-root" ".")
                (println))}
```

Reply:
We can wait for the next release, no problem
About the unused-public-var, we could use `#_{:clj-kondo/ignore [:clojure-lsp/unused-public-var]}`